crates/state/src/account_info.rs

```diff
@@ -32,9 +32,7 @@ impl Default for AccountInfo {
 
 impl PartialEq for AccountInfo {
     fn eq(&self, other: &Self) -> bool {
-        self.balance == other.balance
-            && self.nonce == other.nonce
-            && self.code_hash == other.code_hash
+        (self.balance, self.nonce, self.code_hash) == (other.balance, other.nonce, other.code_hash)
```

Comment:
this also technically copies into a tuple type, I don't think this is the same assembly, so we should keep this unchanged as well